### PR TITLE
[tzlaine-parser] add new port

### PR DESCRIPTION
**Pull request by jiayuehua**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/tzlaine-parser/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,22 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO tzlaine/parser
+    REF 9b03f3c1f9402d33807b68badd6d8219a11359d1
+    SHA512 94475821acac20f89adcdb6079875f9476b1da46f3f9ddfbe36de08b87513bbc768921d8e6b801326f25c8185b974a2a153bfb66654f31696bf935b3cd24dfcb
+    HEAD_REF main
+    PATCHES add-install-configuration.patch
+)
+message (FATAL "not finished")
```
Comment (MonicaLiu0311) [inline ports/tzlaine-parser/portfile.cmake:9]:
```suggestion
```

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-windows` (header files found):
```
tzlaine-parser provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(unofficial-tzlaine-parser CONFIG REQUIRED)
  target_link_libraries(main PRIVATE unofficial::tzlaine-parser::parser)
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Tagging vcpkg-team-review because this installs its headers to `boost/parser`

File: ports/tzlaine-parser/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "tzlaine-parser",
+  "version-date": "2024-10-04",
+  "description": "A C++ parser combinator library.",
+  "homepage": "https://tzlaine.github.io/parser/doc/html/index.html",
```
Comment (BillyONeal) [inline ports/tzlaine-parser/vcpkg.json:5]: This link is broken.

**Comment by BillyONeal** [comment]
Upstream says:

>This library first appeared in Boost 1.87.0

Should we wait for boost 1.87 so that this port doesn't stomp on Boost 1.87 when we take that?

**Comment by jiayuehua** [comment]
I think wait for boost 1.87 better.

File: ports/tzlaine-parser/vcpkg.json (hunk 2 above)
Comment (jiayuehua) [reply ports/tzlaine-parser/vcpkg.json:5]: very recently broken.

**Pull request closed without merging**